core/store/src/trie/shard_tries.rs

```diff
@@ -101,16 +107,27 @@ impl ShardTries {
                 .or_insert_with(|| self.0.trie_cache_factory.create_cache(&shard_uid))
                 .clone()
         };
-        let store = Box::new(TrieCachingStorage::new(self.0.store.clone(), cache, shard_uid));
-        Trie::new(store)
+        let storage = Box::new(TrieCachingStorage::new(self.0.store.clone(), cache, shard_uid));
+        let flat_state =
+            if use_flat_state { Some(FlatState { store: self.0.store.clone() }) } else { None };
+        Trie::new(storage, flat_state)
+    }
+
+    pub fn get_trie_with_optional_flat_state_for_shard(
```

Comment:
Can we make this private? There are two reasons why `pub` doesn't sit well with me: 

* functions with a bunch of switches (like two bool parametres here) are usually brittle  public interface
* before, we didn't expose `is_view` as a switch -- the caller always statically knew whether they need view trie or plain trie. It feels better not to expose this, as this is bug prone. In fact, it seems to me that, at the call-site, we actually pass wrong `is_view`?

Reply:
Yeah, it was a mistake.
We need tries (states) of three types: view trie, trie, trie with flat state, so now I create 3 separate methods for getting them.